src/main/java/com/docutools/poipath/xwpf/XWPFBuilder.java

```diff
@@ -0,0 +1,215 @@
+package com.docutools.poipath.xwpf;
+
+import java.util.Arrays;
+import java.util.function.Consumer;
+import org.apache.poi.xwpf.usermodel.IBodyElement;
+import org.apache.poi.xwpf.usermodel.XWPFDocument;
+import org.apache.poi.xwpf.usermodel.XWPFParagraph;
+import org.apache.poi.xwpf.usermodel.XWPFRun;
+import org.apache.poi.xwpf.usermodel.XWPFTable;
+import org.apache.poi.xwpf.usermodel.XWPFTableCell;
+import org.apache.poi.xwpf.usermodel.XWPFTableRow;
+
+/**
+ * Provides a simple builder pattern for defining XWPF data structures.
+ *
+ * @param <T> the XWPF type {@link XWPFParagraph}, {@link XWPFRun}, {@link XWPFTable}, ...
+ */
+public record XWPFBuilder<T>(XWPFBuilder<?> parent, T bodyElement) {
+
+  /**
+   * Applies a given function on the XWPF body element.
+   *
+   * @return {@code this}
+   */
+  public XWPFBuilder<T> modify(Consumer<T> modifier) {
+    modifier.accept(bodyElement);
+    return this;
+  }
+
+  /**
+   * Creates a new {@link XWPFParagraph}
+   *
+   * @return the {@link XWPFBuilder} for {@link XWPFParagraph}
+   */
+  public XWPFBuilder<XWPFParagraph> paragraph() {
+    if (bodyElement instanceof XWPFDocument document) {
+      return new XWPFBuilder<>(this, document.createParagraph());
+    }
+    if (bodyElement instanceof XWPFTableCell cell) {
+      return new XWPFBuilder<>(this, cell.addParagraph());
+    }
+    throw isUnsupported(XWPFParagraph.class);
+  }
+
+  /**
+   * Creates a new {@link XWPFParagraph} with the given text
+   *
+   * @param text paragraph content
+   * @return {@code this}
+   */
+  public XWPFBuilder<T> paragraph(String text) {
+    var paragraphTexts = text.split("\n");
```

Comment:
Do we really need to create one paragraph per line?

Reply:
Nope, fixed.